### PR TITLE
Fix #17497; moves bardrone from map to drone types

### LINKED ISSUE #17497: Drones have lost their autogenerated number
My fault, I'll fix it

### PR DESCRIPTION
**Pull request by coiax**
Fixes #17497.

Bardrone is now a valid type, rather than map specific.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm [hunk 1]
```diff
@@ -54,6 +54,19 @@
 	..()
 	desc += " This drone appears to have a complex holoprojector built on its 'head'."
 
+/mob/living/simple_animal/drone/snowflake/bardrone
+	name = "Bardrone"
+	desc = "A barkeeping drone, an indestructible robot built to tend bars."
+	health = 3000
+	maxHealth = 3000
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm]: Health changes aren't needed with GODMODE

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm [hunk 2]
```diff
@@ -62,10 +62,11 @@
...
 	var/hacked = 0 //If we have laws to destroy the station
 	var/datum/personal_crafting/handcrafting
+	var/has_number = TRUE
 
 /mob/living/simple_animal/drone/New()
 	..()
-	if(name != initial(name))
+	if(has_number)
 		name = name + " ([rand(100,999)])"
 	real_name = name
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm]: This entire New() is unneeded if you use the `unique_name` var (see living.dm)

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm [hunk 3]
```diff
@@ -62,10 +62,11 @@
 	var/visualAppearence = MAINTDRONE //What we appear as
 	var/hacked = 0 //If we have laws to destroy the station
 	var/datum/personal_crafting/handcrafting
+	var/has_number = TRUE
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm]: Similarly, this var is also unneeded if using `unique_name`

File: _maps/shuttles/emergency_bar.dmm [hunk 4]
```diff
@@ -34,7 +34,7 @@
 "aH" = (/obj/machinery/door/airlock/shuttle{name = "Emergency Shuttle Airlock"},/obj/docking_port/mobile/emergency{name = "The Emergency Escape Bar"},/turf/open/floor/plasteel/bar,/area/shuttle/escape)
 "aI" = (/obj/structure/table,/obj/machinery/chem_dispenser/drinks/beer,/turf/open/floor/plasteel/bar,/area/shuttle/escape)
 "aJ" = (/obj/structure/chair,/turf/open/floor/plasteel/bar,/area/shuttle/escape)
-"aK" = (/mob/living/simple_animal/drone/snowflake{desc = "A barkeeping drone, an indestructible robot built to tend bars."; dir = 8; health = 3000; languages = 127; laws = "1. Serve drinks. 2. Talk to patrons. 3. Don't get messed up in their affairs."; maxHealth = 3000; name = "Bardrone"; seeStatic = 0},/turf/open/floor/plasteel/bar,/area/shuttle/escape)
+"aK" = (/mob/living/simple_animal/drone/snowflake/bardrone,/turf/open/floor/plasteel/bar,/area/shuttle/escape)
```
Comment (ChangelingRain) [inline _maps/shuttles/emergency_bar.dmm]: Shouldn't this be a bardrone shell, instead? otherwise, it needs an admin to put someone in

Comment (coiax) [inline _maps/shuttles/emergency_bar.dmm]: That's intended. It's an indestructible drink serving machine, it's admin opt in.

**Comment by optimumtact** [comment]
code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm:60:error: has_number:  undefined var

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm [hunk 5]
```diff
@@ -65,9 +65,6 @@
 
 /mob/living/simple_animal/drone/New()
 	..()
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm]: You can remove this!

File: code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm [hunk 6]
```diff
@@ -54,6 +54,18 @@
...
+	name = "Bardrone"
+	desc = "A barkeeping drone, an indestructible robot built to tend bars."
+	has_number = FALSE
+	seeStatic = FALSE
+	laws = "1. Serve drinks.\n\
+		2. Talk to patrons.\n\
+		3. Don't get messed up in their affairs."
+	languages = ALL
+	status_flags = GODMODE // Please don't punch the barkeeper
+	unique_name = TRUE
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm]: Needs to be FALSE for it to not generate a number, you gotta put unique_name = TRUE on the base drone

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm (hunk 5 above)
Comment (coiax) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm]: Um, but don't we want /mob/living/New() to run?

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm [hunk 7]
```diff
@@ -64,10 +65,7 @@
 	var/datum/personal_crafting/handcrafting
 
 /mob/living/simple_animal/drone/New()
-	..()
-	if(name != initial(name))
-		name = name + " ([rand(100,999)])"
-	real_name = name
+	. = ..()
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm]: No, really, you can remove these two lines you don't *need* the new if this is all it's doing

**Pull request merged**